examples/server/tests/features/steps/steps.py

```diff
@@ -1223,6 +1276,15 @@ def start_server_background(context):
         'stdout': subprocess.PIPE,
         'stderr': subprocess.PIPE
     }
+
+    # Shut down previous server if there is one:
```

Comment:
Server is supposed to be stopped in the after_scenario hook. Why this is now necessary?

Reply:
I used this in the previous version of the tests where I varied the number of slots rather than the number of completion requests. Though this is not done in any of the current tests, starting the server twice without this will cause the tests to hang indefinitely. So I figured it would be better to keep it.

Reply:
It was present in the first version, but we feel it adds confusion. But you can move it in the start scenario hook, it would be clearer ?

Reply:
I think the simplest solution would be to just remove it since right now it's not needed.